### PR TITLE
Project5 controller test

### PR DESCRIPTION
**Pull request by IamYunusAli**
- Add RSpec as a gem to your project.
- Create a Request spec file for all your controllers.
- Make sure that for each action you check:
    - If the response status was correct.
    - If a correct template was rendered.
    - If the response body includes correct placeholder text.
 
- use `rspec spec/requests` to run only the controller rspec test

control test

### CODE REVIEW

File: spec/requests/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                    likes_counter: 2)
+    get "/users/#{user.id}/posts/#{post.id}"
+  end
+
+  it 'response status was correct' do
+    expect(response).to have_http_status(:ok)
+  end
+
+  it 'correct template was rendered' do
+    expect(response).to render_template(:index)
```
Comment (Davidosky007) [inline spec/requests/post_spec.rb]: * As stated in [the requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=Make%20sure%20that%20for,includes%20correct%20placeholder%20text.) Kindly write a test for each action in your controllers ( which are `index` and `show` ) but here all tests are written for the index action only. kindly add a test for show action to tell If a correct template was rendered.

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @IamYunusAli 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* Well written PR description 👍 
* No linter errors 👍 
* Descriptive commit messages 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShadyShawkat — APPROVED** [review]
## Hi @IamYunusAli, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- All tests are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/post_spec.rb (hunk 1 above)
Comment (IamYunusAli) [reply spec/requests/post_spec.rb]: Totally forgot the make the changes from the index to show. when I run the test when it passes I thought It was working. fixed it now. @Davidosky007 thank you for pointing that out.

**Pull request merged**